README.md

```diff
@@ -256,253 +220,32 @@ This is a **huge improvement**, because:
 
   * If we change the storage, the callers aren't affected.
 
-Here is an extended example of a repository that uses the SQL adapter.
-
-```ruby
-class ArticleRepository
-  include Hanami::Repository
-
-  def self.most_recent_by_author(author, limit = 8)
-    query do
-      where(author_id: author.id).
-        desc(:id).
-        limit(limit)
-    end
-  end
-
-  def self.most_recent_published_by_author(author, limit = 8)
-    most_recent_by_author(author, limit).published
-  end
-
-  def self.published
-    query do
-      where(published: true)
-    end
-  end
+### Mapping
 
-  def self.drafts
-    exclude published
-  end
+Hanami::Model can **_automap_** columns from relations and entities attributes.
 
-  def self.rank
-    published.desc(:comments_count)
-  end
-
-  def self.best_article_ever
-    rank.limit(1)
-  end
-
-  def self.comments_average
-    query.average(:comments_count)
-  end
-end
-```
-
-You can also extract the common logic from your repository into a module to reuse it in other repositories. Here is a pagination example:
-
-```ruby
-module RepositoryHelpers
-  module Pagination
-    def paginate(limit: 10, offset: 0)
-      query do
-        limit(limit).offset(offset)
-      end
-    end
-  end
-end
-
-class ArticleRepository
-  include Hanami::Repository
-  extend RepositoryHelpers::Pagination
-
-  def self.published
-    query do
-      where(published: true)
-    end
-  end
-
-  # other repository-specific methods here
-end
-```
-
-That will allow `.paginate` usage on `ArticleRepository`, for example:
-`ArticleRepository.published.paginate(15, 0)`
-
-**Your models and repositories have to be in the same namespace.** Otherwise `Hanami::Model::Mapper#load!`
-will not initialize your repositories correctly.
-
-```ruby
-class MyHanamiApp::Model::User
-  include Hanami::Entity
-  # your code here
-end
-
-# This repository will work...
-class MyHanamiApp::Model::UserRepository
-  include Hanami::Repository
-  # your code here
-end
-
-# ...this will not!
-class MyHanamiApp::Repository::UserRepository
-  include Hanami::Repository
-  # your code here
-end
-```
-
-### Data Mapper
-
-A persistence mapper that keeps entities independent from database details.
-It is database independent, it can work with SQL, document, and even with key/value stores.
-
-The role of a data mapper is to translate database columns into the corresponding attribute of an entity.
+However, there could be cases where there is a mismatch between the columns and attributes names.
+This is the case of **legacy databases**.
 
```

Comment:
@beauby Thanks for the suggestion: https://github.com/hanami/model/pull/334/commits/cb59e7e6d8e15f56cac203e2d1948aa686d465d0